### ISSUE TITLE
False Negative NullableToStringCall

### ISSUE THREAD

**Issue by scottkennedy** [detekt/detekt#4988]
## Expected Behavior
After enabling `NullableToStringCall`, I expect it to work.

## Observed Behavior
If I create a file that has nothing other than the "non-compliant" section from https://github.com/detekt/detekt/blob/main/detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/NullableToStringCall.kt

```kt
fun foo(a: Any?): String {
    return a.toString()
}

fun bar(a: Any?): String {
    return "$a"
}
```

I expect errors.

However, all I get is:
```
.../src/main/kotlin/example/Example.kt:1:1: The file does not contain a package declaration. [MissingPackageDeclaration]
.../src/main/kotlin/example/Example.kt:2:5: Functions with exact one statement, the return statement, can be rewritten with ExpressionBodySyntax. [ExpressionBodySyntax]
.../src/main/kotlin/example/Example.kt:6:5: Functions with exact one statement, the return statement, can be rewritten with ExpressionBodySyntax. [ExpressionBodySyntax]
```

If I add a package declaration and change them to expression body syntax, everything passes.

## Steps to Reproduce
1. Use a detekt config file with:
```
...
potential-bugs:
  active: true
...
  NullableToStringCall:
    active: true
...
```

2. Create a file as outlined above.
3. Run Detekt

## Context
We've been incorrectly sending `"null"` strings through our API.  I was going to make an example, but the default example seems to exhibit the same problem.

## Your Environment
* Version of detekt used: 1.20.0 (same issue with 1.21.0-RC1)
* Version of Gradle used (if applicable): 7.4.2
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: macOS 12.4, and whatever Linux distro GitHub Actions uses
* Link to your project (if it's a public repository):

**Comment by cortinico** [comment]
> Gradle scan link (add `--scan` option when running the gradle task):

Can you provide a Gradle Scan? Are you running with Type Resolution also?

**Comment by scottkennedy** [comment]
Ah, my bad, it was not running with type resolution.  It works great now, thanks.
